mabwiser/mab.py

```diff
@@ -413,6 +411,68 @@ def _validate(self):
             check_true(self.n_clusters >= 2, ValueError("The number of clusters must be at least two."))
             check_true(isinstance(self.is_minibatch, bool), TypeError("The is_minibatch flag must be a boolean."))
 
+    class LSHNearest(NamedTuple):
+        """LSH Approximate Nearest Neighbors Policy.
+
+        LSHNearest is a nearest neighbors approach that uses locality sensitive hashing with a simhash to
+        select observations to be used with a learning policy.
+
+        For the simhash, contexts are projected onto an n-dimensional plane and each dimension is evaluated for its
+        sign. This is converted to a base 2 integer used as the hash value to assign the context to a hash table. This
+        process is repeated for a specified number of hash tables, where each has a unique, randomly-generated plane.
+        To select the neighbors for a context, the hash value is calculated for each hash table and any contexts with
+        the same hashes are selected as the neighbors.
+
+        As with the radius or k value for other nearest neighbors algorithms, selecting the best number of dimensions
+        and tables requires tuning. For the dimensions, a good starting point is to use the log of the square root of
+        the number of rows in the training data. This will give you sqrt(n_rows) number of hashes. The number of
+        dimensions and number of tables have inverse effects on the number of empty neighborhoods and average
```

Comment:
can you confirm: based on n_dim, and n_tables, if you increase them, it is less approx and more exact? 
What would be the extreme case look like to get "exact" nhoods (and sure if that's possible though)

Reply:
Agreed with Serdar, making the effect of these params explicit would be very helpful.

Reply:
Simhash uses an approximation of cosine similarity, so I don't think there is a way to guarantee that you can get the same results as with Radius/KNearest with a cosine similarity metric. Increasing n_dimensions decreases the number of collisions for a given hyperplane, so the neighbors are more "exact", while increasing n_tables increases the diversity of the planes used and so finds neighbors the others miss. One thing to note is that this is O(2^n_dimensions) so some imprecision in the neighbors needs to be acceptable, otherwise you would be better off using Radius or KNearest.

Reply:
cool, thanks for the comments. You can incorporate some of that to the docs in the code, so that reader has a direction on how to turn the knobs  on these params between approx -- exact spectrum,.